File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque
def solution(prices):
    answer = []
    prices = deque(prices)
    while prices:
        time = 0
        x = prices.popleft()
        for price in prices:
            time += 1
            if x > price:
                break
        answer.append(time)
        
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    queue<int> q;
    int time;
    int idx = 0;
    for (int i = 0; i < prices.size() ; i++){
        q.push(prices[i]);
    }
    while (q.size()>0) {
        int cur = q.front();
        q.pop(); 
        time = 0;
        // cout << cur << endl;
        for (int i = idx; i < prices.size() ; i++){
            // cout << cur << ":" << time << endl;
            time++;

            if (cur > prices[i]) {
                break;
            }
        }
        idx++;
        answer.push_back(time-1);
        }
    return answer;
}